src/mono/wasm/debugger/BrowserDebugProxy/ValueTypeClass.cs

```diff
@@ -110,17 +111,39 @@ JObject GetFieldWithMetadata(FieldTypeClass field, JObject fieldValue, bool isSt
                 return fieldValue;
             }
         }
+        public async Task<string> ToString(MonoSDBHelper sdbAgent, CancellationToken token)
+        {
+            var typeInfo = await sdbAgent.GetTypeInfo(TypeId, token);
+            if (typeInfo == null)
+                return null;
+            if (typeInfo.Name == "object")
+                return null;
+            Microsoft.WebAssembly.Diagnostics.MethodInfo methodInfo = typeInfo.Info.Methods.FirstOrDefault(m => m.Name == "ToString");
+            if (!IsEnum && methodInfo == null)
+                return null;
+            int[] methodIds = await sdbAgent.GetMethodIdsByName(TypeId, "ToString", token);
+            if (methodIds == null)
+                return null;
+            try {
+                var retMethod = await sdbAgent.InvokeMethod(Buffer, methodIds[0], token, "methodRet");
+                return retMethod["value"]?["value"].Value<string>();
+            }
+            catch (Exception e)
+            {
+                sdbAgent.logger.LogDebug($"Error while evaluating ToString method: {e}");
+            }
+            return null;
+        }
 
         public async Task<JObject> ToJObject(MonoSDBHelper sdbAgent, bool forDebuggerDisplayAttribute, CancellationToken token)
         {
             string description = className;
             if (ShouldAutoInvokeToString(className) || IsEnum)
             {
-                int[] methodIds = await sdbAgent.GetMethodIdsByName(TypeId, "ToString", token);
-                if (methodIds == null)
+                var toString = await ToString(sdbAgent, token);
+                if (toString == null)
                     throw new InternalErrorException($"Cannot find method 'ToString' on typeId = {TypeId}");
```

Comment:
Also, the error isn't correct because `toString` could be null because we couldn't find it, or it wasn't applicable, or invoking the method failed with an exception.

Reply:
changed.